processor/src/main/java/org/mapstruct/ap/internal/model/source/selector/MethodSelection.java

```diff
@@ -16,28 +16,40 @@
  *  See the License for the specific language governing permissions and
  *  limitations under the License.
  */
-package org.mapstruct.ap.internal.model;
+package org.mapstruct.ap.internal.model.source.selector;
 
 import java.util.List;
 
 import org.mapstruct.ap.internal.model.common.Parameter;
 import org.mapstruct.ap.internal.model.source.Method;
 
 /**
- * Represents a reference to a factory method.
+ * A selected method with additional metadata that might be required for further usage of the selected method.
  *
- * @author Remo Meier
+ * @author Andreas Gudian
  */
-public class ObjectFactoryMethodReference extends MethodReference {
+public class MethodSelection<T extends Method> {
+    private T method;
+    private List<Parameter> parameterBindings;
 
-    private final List<Parameter> parameterAssignments;
+    public MethodSelection(T method) {
+        this.method = method;
+    }
+
+    public T getMethod() {
+        return method;
+    }
+
+    public List<Parameter> getParameterBindings() {
```

Comment:
That holds the information about what parameters/arguments were successfully matched for the method call. When rendering the method call in the templates, that information is required to know what to put where (e.g. a source parameter of the mapping method, the target type class literal, the mapping target, or an assignment -- or later on any mapping context parameters).